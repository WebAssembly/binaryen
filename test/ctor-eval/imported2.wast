(module
  (memory 256 256)
  (data (i32.const 10) "waka waka waka waka waka")
  ;; stack imports are special-cased
  (import "env" "STACKTOP" (global $STACKTOP i32))
  (import "env" "STACK_MAX" (global $STACK_MAX i32))
  ;; other imports must not be touched!
  (import "env" "tempDoublePtr" (global $tempDoublePtr i32))
  (export "test1" $test1)
  (export "test2" $test2)
  (export "test3" $test3)
  ;; ok to modify a global, if we keep it the same value
  (global $mine (mut i32) (i32.const 1))
  (func $test1
    (set_global $mine (i32.const 2))
    (set_global $mine (i32.const 1)) ;; restore!
    (i32.store8 (i32.const 12) (i32.const 115))
  )
  (func $test2
    (set_global $mine (i32.const 2)) ;; embadden
    (i32.store8 (i32.const 13) (i32.const 115))
  )
  (func $test3
    (i32.store8 (i32.const 14) (i32.const 115))
  )
)
