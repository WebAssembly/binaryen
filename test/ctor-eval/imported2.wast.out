(module
 (type $0 (func))
 (import "env" "STACKTOP" (global $STACKTOP i32))
 (import "env" "STACK_MAX" (global $STACK_MAX i32))
 (import "env" "tempDoublePtr" (global $tempDoublePtr i32))
 (global $mine (mut i32) (i32.const 1))
 (memory $0 256 256)
 (data (i32.const 10) "wasa waka waka waka waka")
 (export "test1" (func $test1))
 (export "test2" (func $test2))
 (export "test3" (func $test3))
 (func $test1 (type $0)
  (nop)
 )
 (func $test2 (type $0)
  (set_global $mine
   (i32.const 2)
  )
  (i32.store8
   (i32.const 13)
   (i32.const 115)
  )
 )
 (func $test3 (type $0)
  (i32.store8
   (i32.const 14)
   (i32.const 115)
  )
 )
)
